go-sdk/internal/sparse-merkle-tree/smt/merkletree.go

```diff
@@ -240,7 +262,8 @@ func (mt *sparseMerkleTree) addLeaf(newLeaf core.Node, currentNodeIndex core.Nod
 			return nil, err
 		}
 		// persist the updated branch node
-		return mt.addNode(newBranchNode)
+		log.WithLogField("level", strconv.Itoa(level)).Debugf("Inserting new branch node %s (leftChild=%s, rightChild=%s)", newBranchNode.Ref().Hex(), newBranchNode.LeftChild().Hex(), newBranchNode.RightChild().Hex())
```

Comment:
doing the logging here rather than the DB client is mainly to have the right context information, in particular the `level` which is useful for debugging purposes.